packages/button/src/component.tsx

```diff
@@ -28,25 +28,79 @@ export const Button = forwardRef<
+    buttonUtilityFlat:
+      'bg-transparent border-0 font-bold max-w-max focusable justify-center transition-colors ease-in-out i-text-$color-button-utility-quiet-text i-bg-$color-button-utility-quiet-background hover:i-bg-$color-button-utility-quiet-background-hover rounded-4', // .button--utility-flat
+    buttonPill:
+      'font-bold max-w-max focusable justify-center transition-colors ease-in-out rounded-full! border-0! p-4 i-text-$color-button-pill-icon hover:i-text-$color-button-pill-icon-hover active:i-text-$color-button-pill-icon-active i-bg-$color-button-pill-background hover:i-bg-$color-button-pill-background-hover active:i-bg-$color-button-pill-background-active inline-flex items-center justify-center hover:bg-clip-padding', // .button--pill
+    buttonLink:
+      'max-w-max bg-transparent focusable ease-in-out inline i-text-$color-button-link-text active:underline hover:underline', //.button--link /* Buttons pretending to be links, (Should rather inherit the actual link setup in the future?)  */
+    // Font stuff
+    buttonMediumFontSize: 'leading-[24] text-m', // .button--secondary
+    buttonSecondaryPadding: 'py-10 px-14', // .button--secondary.button--small
+    buttonMediumPadding: 'py-12 px-16', // .button--primary, .button--quiet, .button--destructive, .button--destructive-flat, .button--utility-flat
+    buttonUtilitySizing: 'px-[15px] py-[11px] font-bold', // .button--utility
+    buttonPillSizing: 'min-h-[44px] min-w-[44px]', // .button--utility
+    buttonSmallSecondary: 'px-16 py-6 text-xs', // .button--small
+    buttonSmallPrimary: 'px-16 py-8 text-xs', // .button--small.button--primary, .button--small.button--destructive, .button--small.button--destructive-flat, .button--small.button--order, .button--small.button--quiet
+    buttonSmallUtility: 'px-[15px] py-[7px] text-xs', // .button--small.button--utility
+    buttonSmallButtonPill: 'min-h-32 min-w-32 text-xs', // .button--small.button--pill
+    buttonSmallButtonLink: 'p-0 text-xs', // .button--small.button--link
+    // Disabled
+    buttonIsDisabled:
+      'font-bold max-w-max justify-center transition-colors ease-in-out i-bg-$color-button-disabled-background! i-text-$color-button-disabled-text cursor-default pointer-events-none', // .button:disabled, .button--is-disabled
+    // Progress indicator
+    buttonInProgress:
+      'border-transparent! animate-inprogress i-text-$color-button-loading-text! pointer-events-none i-bg-$color-button-loading-background!', // .button--in-progress, a.button--in-progress:visited
+    linkAsButton: 'inline-block hover:no-underline',
+    a11y: 'sr-only',
+  };
+
+  const buttonSecondaryDefault = secondary && !quiet || !buttonTypes.find(b => !!props[b]);
+  const buttonDestructive = negative && !quiet;
+  const buttonFlat = secondary && quiet;
+  const buttonDestructiveFlat = negative && quiet;
+  const buttonUtilityFlat = utility && quiet;
```

Comment:
Not saying that we need to fix this now, but I don't like that we are using both destructive/negative and flat/quiet, it's a bit confusing 😅 Would be nice to call the button class e.g. `buttonNegative` and `buttonQuiet` to match the props the user is setting.

Reply:
yes, totally agree. We have discussed it before. We cannot do that now as shortcuts are used with the old names and changes will bring breaking changes. And in the beginning of the project we said, or at least I want to remember that, that no breaking changes were allowed. Maybe I don't remember the details and maybe it is doable but it should for sure be done when possible